File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        
        for (int num : nums) {
            set.add(num);
        }
        
        int res = 0;
        for (int num : set) {
            if (!set.contains(num-1)) {
                int currentNum = num;
                int currentStreak = 1;
                while (set.contains(currentNum+1)) {
                    currentStreak++;
                    currentNum++;
                }
                res = Math.max(res, currentStreak);
            }
        }
        return res;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        
        longestStreak = 0;
        
        for num in nums:
            if (num-1) not in numSet:
                currentNum = num
                streak = 0
                while (currentNum in numSet):
                    streak += 1
                    currentNum += 1    
                longestStreak = max(longestStreak, streak)
        return longestStreak   